commit 86f5a6f7ca4ba104979d1904e96c34999634c63d
Author: Sergei Poselenov <sposelenov@emcraft.com>
Date:   Wed Sep 19 16:18:56 2018 +0300

    RM 2683. Disabled workaround for the 1N14W mask Industrial chips. These chips
    will not be used for production.

diff --git a/plat/freescale/imx8mq/gpc.c b/plat/freescale/imx8mq/gpc.c
index 4e8f7bac..3a9026f7 100644
--- a/plat/freescale/imx8mq/gpc.c
+++ b/plat/freescale/imx8mq/gpc.c
@@ -579,8 +579,18 @@ static void imx_gpc_pm_domain_enable(uint32_t domain_id, uint32_t on)
 	val = 1 << (domain_id > 3 ? (domain_id + 3) : domain_id);
 	mmio_write_32(reg, val);
 	count = 100;
+#if 0
+	/* This change allows Linux to boot all the way on the i.MX 8M chips
+	 * with the 1N14W mask. Otherwise, these chips get stuck in the Linux
+	 * USB code waiting indefinitely for the USB_OTG1_SW_PUP_REQ bit to clear.
+	 */
 	while((mmio_read_32(reg) & val) && count--)
 		;
+#else
+	while(mmio_read_32(reg) & val)
+		;
+#endif
+
 	imx_gpc_set_m_core_pgc(gpc_pu_m_core_offset[domain_id], false);
 }
 
